### PR TITLE
Add classification to all authtoken fields

### PR DESCRIPTION
**Pull request by johanbrandhorst**
For https://hashicorp.atlassian.net/browse/ICU-3915

### CODE REVIEW

**Review by tmessi — APPROVED** [review]

**Review by jimlambrt — APPROVED** [review]

**Comment by jimlambrt** [comment]
to be clear, I only reviewed the protos classifications.

File: internal/proto/controller/api/resources/authtokens/v1/authtoken.proto [hunk 1]
```diff
@@ -10,38 +10,38 @@ import "controller/api/resources/scopes/v1/scope.proto";
...
-	string id = 10;
+	string id = 10;  // @gotags: `class:"public"`
 
 	// The Scope in which this Auth Token was generated.
-	string scope_id = 20 [json_name="scope_id"];
+	string scope_id = 20 [json_name="scope_id"];  // @gotags: `class:"public"`
 
 	// Output only. Scope information for this resource.
-	resources.scopes.v1.ScopeInfo scope = 30;
+	resources.scopes.v1.ScopeInfo scope = 30;  // @gotags: `class:"public"`
```
Comment (jefferai) [inline internal/proto/controller/api/resources/authtokens/v1/authtoken.proto:19]: This doesn't need a tag

File: internal/proto/controller/api/resources/authtokens/v1/authtoken.proto [hunk 2]
```diff
@@ -10,38 +10,38 @@ import "controller/api/resources/scopes/v1/scope.proto";
...
-	string auth_method_id = 60 [json_name="auth_method_id"];
+	string auth_method_id = 60 [json_name="auth_method_id"];  // @gotags: `class:"public"`
 
 	// Output only. The ID of the Account associated with this Auth Token.
-	string account_id = 70 [json_name="account_id"];
+	string account_id = 70 [json_name="account_id"];  // @gotags: `class:"public"`
 
 	// Output only. The time this resource was created.
-	google.protobuf.Timestamp created_time = 80 [json_name="created_time"];
+	google.protobuf.Timestamp created_time = 80 [json_name="created_time"];  // @gotags: `class:"public"`
```
Comment (jefferai) [inline internal/proto/controller/api/resources/authtokens/v1/authtoken.proto:34]: Same questions as in #2004 around proto standard types.

File: internal/proto/controller/api/resources/scopes/v1/scope.proto [hunk 3]
```diff
@@ -35,7 +35,7 @@ message Scope {
   string scope_id = 20 [json_name = "scope_id"];  // @gotags: `class:"public"`
 
   // Output only. Scope information for this resource.
-  ScopeInfo scope = 30;
+  ScopeInfo scope = 30;  // @gotags: `class:"public"`
```
Comment (jefferai) [inline internal/proto/controller/api/resources/scopes/v1/scope.proto:38]: This doesn't need a tag

File: internal/proto/controller/api/resources/scopes/v1/scope.proto [hunk 4]
```diff
@@ -69,5 +69,5 @@ message Scope {
   repeated string authorized_actions = 300 [json_name = "authorized_actions"];  // @gotags: `class:"public"`
 
   // Output only. The authorized actions for the scope's collections.
-  map<string, google.protobuf.ListValue> authorized_collection_actions = 310 [json_name = "authorized_collection_actions"];
+  map<string, google.protobuf.ListValue> authorized_collection_actions = 310 [json_name = "authorized_collection_actions"];  // @gotags: `class:"public"`
```
Comment (jefferai) [inline internal/proto/controller/api/resources/scopes/v1/scope.proto:72]: How does this get handled?

File: internal/proto/controller/api/services/v1/authtokens_service.proto [hunk 5]
```diff
@@ -49,25 +49,25 @@ service AuthTokenService {
...
-  bool recursive = 20 [json_name="recursive"];
-  string filter = 30 [json_name="filter"];
+  string scope_id = 1 [json_name="scope_id"];  // @gotags: `class:"public"`
+  bool recursive = 20 [json_name="recursive"];  // @gotags: `class:"public"`
+  string filter = 30 [json_name="filter"];  // @gotags: `class:"public"`
 }
 
 message ListAuthTokensResponse {
-  repeated resources.authtokens.v1.AuthToken items = 1;
+  repeated resources.authtokens.v1.AuthToken items = 1;  // @gotags: `class:"public"`
```
Comment (jefferai) [inline internal/proto/controller/api/services/v1/authtokens_service.proto:66]: This doesn't need a tag

File: internal/proto/controller/api/resources/scopes/v1/scope.proto (hunk 4 above)
Comment (johanbrandhorst) [reply internal/proto/controller/api/resources/scopes/v1/scope.proto:72]: Yeah this won't work

File: internal/proto/controller/api/services/v1/authtokens_service.proto (hunk 5 above)
Comment (johanbrandhorst) [reply internal/proto/controller/api/services/v1/authtokens_service.proto:66]: According to @tmessi we _do_ need it on repeated fields?

Comment (tmessi) [reply internal/proto/controller/api/services/v1/authtokens_service.proto:66]: I'm pretty sure I tried to remove these any my tests started failing with these list not being processed correctly. I can try again though, maybe I messed something up.

Comment (jefferai) [reply internal/proto/controller/api/services/v1/authtokens_service.proto:66]: To me this feels like the same case as a normal message; the fact that it's repeated is beside the point. The tagging will still proceed according to the inner messages. (And if it doesn't, that feels like a problem that needs to be solved.)

Comment (johanbrandhorst) [reply internal/proto/controller/api/services/v1/authtokens_service.proto:66]: Will remove this

File: internal/proto/controller/api/resources/authtokens/v1/authtoken.proto (hunk 2 above)
Comment (johanbrandhorst) [reply internal/proto/controller/api/resources/authtokens/v1/authtoken.proto:34]: Will keep these around, even though they don't have any effect, to make it clear that this field is public

**Pull request merged**